geode-core/src/main/java/org/apache/geode/internal/cache/Oplog.java

```diff
@@ -191,8 +192,8 @@
    */
   private final AtomicLong totalLiveCount = new AtomicLong(0);
 
-  private final ConcurrentMap<Long, DiskRegionInfo> regionMap =
-      new ConcurrentHashMap<>();
+  private final AtomicReference<ConcurrentMap<Long, DiskRegionInfo>> regionMap =
```

Comment:
Maybe it would be better to hide this ConcurrentMap and atomic behind a RegionMap class. It could be a static inner class of Oplog. It looks like we only use a few methods from ConcurrentMap. Your RegionMap class could use an atomic but it could also add a "close" method that you could use when you currently set the atomic to reference an empty map. When you call close you could just set the atomic to null. In all the methods if they see null then they can just behave as if the map is empty. Write ops that see a null map could do nothing or throw an Exception since they really shouldn't happen after close has been called. Read ops can just return what they would on an empty ConcurrentMap. This should use even less memory than a default ConcurrentHashMap and I think it would make the code a bit cleaner.

Reply:
@dschneider-pivotal thank you for the review!

Yes, we agree with you. We have tried to implement what you suggested in commit https://github.com/apache/geode/pull/7145/commits/b0519f48f96dbe04c8cf4c43366694c756570f24 .